modules/demo/rapids-api-server/routes/graphology/index.js

```diff
@@ -123,4 +194,145 @@ module.exports = async function(fastify, opts) {
       }
     }
   });
+
+  fastify.route({
+    method: 'GET',
+    url: '/get_table/:table',
+    schema: {querystring: {table: {type: 'string'}}},
+    handler: async (request, reply) => {
+      let message = 'Error';
+      let result  = {'params': JSON.stringify(request.params), success: false, message: message};
+      const table = gpu_cache.getDataframe(request.params.table);
+      if (table == undefined) {
+        result.message = 'Table not found';
+        reply.code(404).send(result);
+      } else {
+        const writer = RecordBatchStreamWriter.writeAll(table.toArrow());
+        reply.code(200).send(writer.toNodeStream());
+      }
+    }
+  });
+
+  fastify.route({
+    method: 'GET',
+    url: '/nodes/bounds',
+    handler: async (request, reply) => {
+      let message = 'Error';
+      let result  = {success: false, message: message};
+      const df    = gpu_cache.getDataframe('nodes');
+      if (df == undefined) {
+        result.message = 'Table not found';
+        reply.code(404).send(result);
+      } else {
+        // compute xmin, xmax, ymin, ymax
+        const x = df.get('x');
+        const y = df.get('y');
+
+        result.bounds =
+          {xmin: x._col.min(), xmax: x._col.max(), ymin: y._col.min(), ymax: y._col.max()};
+        result.message = 'Success';
+        result.success = true;
+        reply.code(200).send(result);
+      }
+    }
+  });
+
+  fastify.route({
+    method: 'GET',
+    url: '/nodes/',
+    handler: async (request, reply) => {
+      let message = 'Error';
+      let result  = {success: false, message: message};
+      const df    = gpu_cache.getDataframe('nodes');
+      if (df == undefined) {
+        result.message = 'Table not found';
+        reply.code(404).send(result);
+      } else {
+        // tile x, y, size, color
+        let tiled = Series.sequence({type: new Float32, init: 0.0, size: (4 * df.numRows)});
+        let base_offset =
+          Series.sequence({type: new Int32, init: 0.0, size: df.numRows})._col.mul(4);
+        //
+        // Duplicatin the sigma.j createNormalizationFunction here because there's no other way
+        // to let the Graph object compute it.
+        //
+        let x     = df.get('x');
+        let y     = df.get('y');
+        let color = df.get('color');
+        const ratio =
+          Series.new([x._col.max() - x._col.min(), y._col.max() - y._col.min()])._col.max();
```

Comment:
Any reason not to use the `Series` methods for this?

```suggestion
        const ratio = scope(() => {
          const [xMin, xMax] = x.minmax();
          const [yMin, yMax] = y.minmax();
          Math.max(xMax - xMin, yMax - yMin);
        });
```

Reply:
I dug in `Series.ts` for `min` and `max` and somehow missed this. :(